functions/Remove-DbaAgentJobStep.ps1

```diff
@@ -81,61 +80,65 @@ Remove the job step from the job on multiple servers using pipeline
             try {
-                $Server = Connect-SqlInstance -SqlInstance $instance -SqlCredential $SqlCredential
+                $server = Connect-SqlInstance -SqlInstance $instance -SqlCredential $SqlCredential
             }
             catch {
                 Stop-Function -Message "Failure" -Category ConnectionError -ErrorRecord $_ -Target $instance -Continue
             }
 
             foreach ($j in $Job) {
-
+                Write-Message -Level Verbose -Message "Processing job $j"
                 # Check if the job exists
                 if ($Server.JobServer.Jobs.Name -notcontains $j) {
-                    Write-Message -Message "Job $j doesn't exists on $instance" -Level Warning
+                    switch ($Mode) {
+                        'Lazy' {
+                            Write-Message -Level Verbose -Message "Job $j doesn't exists on $instance." -Target $instance
+                        }
+                        'Strict' {
+                            Stop-Function -Message "Job $j doesnn't exist on $instance." -Continue -ContinueLabel main -Target $instance -Category InvalidData
+                        }
+                    }
                 }
                 else {
                     # Check if the job step exists
                     if ($Server.JobServer.Jobs[$j].JobSteps.Name -notcontains $StepName) {
-                        Write-Message -Message "Step $StepName doesn't exist for $job on $instance" -Level Warning
+                        switch ($Mode) {
+                            'Lazy' {
+                                Write-Message -Level Verbose -Message "Step $StepName doesn't exist for $job on $instance." -Target $instance
+                            }
+                            'Strict' {
+                                Stop-Function -Message "Step $StepName doesn't exist for $job on $instance." -Continue -ContinueLabel main -Target $instance -Category InvalidData
+                            }
+                        }
                     }
                     else {
-                        # Get the job step
-                        try {
-                            $JobStep = $Server.JobServer.Jobs[$j].JobSteps[$StepName]
-                        }
-                        catch {
-                            Stop-Function -Message "Something went wrong creating the job step" -Target $JobStep -Continue -ErrorRecord $_
-                        }
-
                         # Execute
                         if ($PSCmdlet.ShouldProcess($instance, "Removing the job step $StepName for job $j")) {
                             try {
-                                Write-Message -Message "Removing the job step $StepName for job $j" -Level Verbose
-
+                                $JobStep = $Server.JobServer.Jobs[$j].JobSteps[$StepName]
+                                Write-Message -Level SomewhatVerbose -Message "Removing the job step $StepName for job $j."
```

Comment:
Should we go ahead and have this one call `Remove-DbaAgentJobStep` ?

Reply:
Not sure what you mean on this one, this is within the Remove-DbaAgentJobStep function.

Reply:
Ah read file name wrong